scripts/macro-usage-report.ts

```diff
@@ -148,15 +151,21 @@ async function writeMarkdownTable(filesByMacro: {
       ...paths.map((path) => formatCell(filterFilesByBase(files, path))),
     ];
 
-    process.stdout.write(`|${cells.map((cell) => ` ${cell} `).join("|")}|\n`);
+    if (deprecatedOnly && deprecatedMacros.includes(macro)) {
+      process.stdout.write(`|${cells.map((cell) => ` ${cell} `).join("|")}|\n`);
+    } else if (!deprecatedOnly) {
+      process.stdout.write(`|${cells.map((cell) => ` ${cell} `).join("|")}|\n`);
+    }
```

Comment:
Instead of making the output conditional, we should check this further above:
```
    if (deprecatedOnly && !deprecatedMacros.includes(macro)) {
      continue;
    }
```